File: UniShare.API/Models/DTOs/Auth/RegisterRequest.cs
namespace UniShare.API.Models.DTOs.Auth;

public class RegisterRequest
{
    public string Email { get; set; } = null!;
    public string Password { get; set; } = null!;
    public string FullName { get; set; } = null!;
    public string? PhoneNumber { get; set; }
}

File: UniShare.APP/lib/features/auth/models/register_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'register_request.g.dart';

@JsonSerializable()
class RegisterRequest {
  final String email;
  final String? phoneNumber;
  final String password;
  final String fullName;

  const RegisterRequest({
    required this.email,
    this.phoneNumber,
    required this.password,
    required this.fullName,
  });

  factory RegisterRequest.fromJson(Map<String, dynamic> json) =>
      _$RegisterRequestFromJson(json);

  Map<String, dynamic> toJson() => _$RegisterRequestToJson(this);
}
